core/cmd/remote_client.go

```diff
@@ -548,3 +555,39 @@ func parseResponse(resp *http.Response) ([]byte, error) {
 	}
 	return b, err
 }
+
+func (cli *Client) checkRemoteBuildCompatibility(lggr logger.Logger, c *clipkg.Context) error {
+	resp, err := cli.HTTP.Get("/v2/build_info")
+	if err != nil {
+		lggr.Warnw("Got error querying for version, remote node version is unknown. CLI may behave in unexpected ways.", "err", err)
+		return nil
+	}
+	b, err := parseResponse(resp)
+	if err != nil {
+		lggr.Warnw("Got error querying for version, remote node version is unknown. CLI may behave in unexpected ways.", "err", err)
+		return nil
+	}
+
+	var remoteBuildInfo map[string]string
+	if err := json.Unmarshal(b, &remoteBuildInfo); err != nil {
+		lggr.Warnw("Got error querying for version, remote node version is unknown. CLI may behave in unexpected ways.", "err", err)
+		return nil
+	}
+	remoteVersion, remoteSha := remoteBuildInfo["version"], remoteBuildInfo["commitSHA"]
+	cliVersion, cliSha := static.Version, static.Sha
+
+	remoteSemverUnset := remoteVersion == "unset" || remoteVersion == "" || remoteSha == "unset" || remoteSha == ""
+	cliRemoteSemverMismatch := remoteVersion != cliVersion || remoteSha != cliSha
+
+	if remoteSemverUnset || cliRemoteSemverMismatch {
+		// Show a warning but allow mismatch if running in dev mode or using bypass-version-check flag
+		if cli.Config.Dev() || c.Bool("bypass-version-check") {
```

Comment:
I'd remove the Dev() check here, seems like its easy enough to manually specify "bypass-version-check" when running on develop